src/ast/passes/codegen_llvm.cpp

```diff
@@ -1082,10 +1082,14 @@ ScopedExpr CodegenLLVM::visit(Call &call)
                                                    call.loc);
       b_.CreateStore(Constant::getNullValue(b_.GetType(map.value_type)), val);
       b_.CreateMapUpdateElem(map.ident, scoped_key.value(), val, call.loc);
-      return ScopedExpr();
+      // Assume delete for these kinds of maps always succeeds.
```

Comment:
Are we sure with this?

Reply:
It's obviously not 100% but probably not something the user can do anything about. These scalar maps only have a single entry so no max-entries issues. Not sure what the other cases would be that would cause this to fail.